sequencer/src/block.rs

```diff
@@ -446,8 +404,8 @@ impl BlockHeader for Header {
         self.payload_commitment
     }
 
-    fn metadata(&self) -> &NMTRoot {
-        &self.transactions_root
+    fn metadata(&self) -> &<Self::Payload as BlockPayload>::Metadata {
+        unimplemented!("TODO");
```

Comment:
I think this should be changed to return the namespace table. Maybe it can just return `()` for now if the VID integration is still in progress. I'm a bit unsure how that change is going to interact with this change. cc @ggutoski

Reply:
VID migration has not yet begun. I don't think it matters much what we do here because it'll change soon in the VID integration.  (At that time it'll return the namespace table.) We cannot return `()` here unless we change the `Metadata` assoc type from `NMTRoot` to `()` and I see no reason to bother with that.

Reply:
Well, the reason would be to remove this `unimplemented!()`. Nothing will work with that there

Reply:
I have a quick fix in c274c49 that should be good enough for now.

Reply:
e0d9f3b4c60d361f96c4228b0c95efce400e2642 should fix it.